wgengine/userspace.go

```diff
@@ -73,6 +75,27 @@ type userspaceEngine struct {
 	// Lock ordering: wgLock, then mu.
 }
 
+// RouterGen is the signature for a function that creates a
+// router.Router.
+type RouterGen func(logf logger.Logf, wgdev *device.Device, tundev tun.Device) (router.Router, error)
+
+type EngineConfig struct {
+	// Logf is the logging function used by the engine.
+	Logf logger.Logf
+	// Tun is the TUN device used by the engine.
+	TUN tun.Device
+	// RouterGen is the function used to instantiate the router.
+	RouterGen RouterGen
+	// ListenPort is the port on which the engine will listen.
+	ListenPort uint16
+	// EchoRespondToAll determines whether incoming ICMP Echo requests
```

Comment:
incoming from tailscale peers - this is again the problem that "incoming" is ambiguous, from tailscaled's perspectives packets are incoming from wireguard and the local kernel.